src/Observer.java

```diff
@@ -2,5 +2,5 @@
 public interface Observer {
     /** Updates a state of the order
```

Comment:
Update the comment as well: it no longer updates the state, but rather reacts to changes
```suggestion
    /** Observes changes of the state of the order
```